res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2415 @@
+        'slicerMode': 0x20,
+        'shiftSlicerMode': 0x6D,
+        'samplerMode': 0x22,
+        'shiftSamplerMode': 0x6F,
+        'longPressSamplerMode': 0x41,
+        'parameterLeftHotCueMode': 0x24,
+        'shiftParameterLeftHotCueMode': 0x01,
+        'parameterLeftRollMode': 0x25,
+        'shiftParameterLeftRollMode': 0x02,
+        'parameterLeftSlicerMode': 0x26,
+        'shiftParameterLeftSlicerMode': 0x03,
+        'parameterLeftSamplerMode': 0x27,
+        'shiftParameterLeftSamplerMode': 0x04,
+        'parameterLeftGroup1Mode': 0x28,
+        'shiftParameterLeftGroup1Mode': 0x05,
+        'parameterLeftGroup2Mode': 0x29,
+        'shiftParameterLeftGroup2Mode': 0x06,
+        'parameterLeftGroup3Mode': 0x2A,
+        'shiftParameterLeftGroup3Mode': 0x07,
+        'parameterLeftGroup4Mode': 0x2B,
+        'shiftParameterLeftGroup4Mode': 0x08,
+        'parameterRightHotCueMode': 0x2C,
+        'shiftParameterRightHotCueMode': 0x09,
+        'parameterRightRollMode': 0x2D,
+        'shiftParameterRightRollMode': 0x7A,
+        'parameterRightSlicerMode': 0x2E,
+        'shiftParameterRightSlicerMode': 0x7B,
+        'parameterRightSamplerMode': 0x2F,
+        'shiftParameterRightSamplerMode': 0x7C,
+        'parameterRightGroup1Mode': 0x30,
+        'shiftParameterRightGroup1Mode': 0x7D,
+        'parameterRightGroup2Mode': 0x31,
+        'shiftParameterRightGroup2Mode': 0x7E,
+        'parameterRightGroup3Mode': 0x32,
+        'shiftParameterRightGroup3Mode': 0x7F,
+        'parameterRightGroup4Mode': 0x33,
+        'shiftParameterRightGroup4Mode': 0x00
+    };
+
+    PioneerDDJSX.illuminationControl = {
+        'loadedDeck1': 0x00,
+        'loadedDeck2': 0x01,
+        'loadedDeck3': 0x02,
+        'loadedDeck4': 0x03,
+        'unknownDeck1': 0x04,
+        'unknownDeck2': 0x05,
+        'unknownDeck3': 0x06,
+        'unknownDeck4': 0x07,
+        'playPauseDeck1': 0x0C,
+        'playPauseDeck2': 0x0D,
+        'playPauseDeck3': 0x0E,
+        'playPauseDeck4': 0x0F,
+        'cueDeck1': 0x10,
+        'cueDeck2': 0x11,
+        'cueDeck3': 0x12,
+        'cueDeck4': 0x13,
+        'djAppConnect': 0x09
+    };
+
+    PioneerDDJSX.valueVuMeter = {
+        '[Channel1]_current': 0,
+        '[Channel2]_current': 0,
+        '[Channel3]_current': 0,
+        '[Channel4]_current': 0,
+        '[Channel1]_enabled': 1,
+        '[Channel2]_enabled': 1,
+        '[Channel3]_enabled': 1,
+        '[Channel4]_enabled': 1
+    };
+
+    // set 32 Samplers as default:
+    engine.setValue("[Master]", "num_samplers", 32);
+
+    // activate vu meter timer for Auto DJ:
+    if (PioneerDDJSX.twinkleVumeterAutodjOn) {
+        PioneerDDJSX.vuMeterTimer = engine.beginTimer(200, "PioneerDDJSX.vuMeterTwinkle()");
+    }
+
+    // send init message to controller:
+    midi.sendSysexMsg(PioneerDDJSX.initSendMsg, PioneerDDJSX.initSendMsg.length);
+    PioneerDDJSX.initTimer = engine.beginTimer(1000, "PioneerDDJSX.initTimerTick()");
```

Comment:
That's strange. As far as I know the only difference is that updating the JS file executes the `shutdown` function before calling `init` again.